src/allotropy/parsers/bmg_mars/bmg_mars_parser.py

```diff
@@ -0,0 +1,270 @@
+            plate_reader_aggregate_document=PlateReaderAggregateDocument(
+                plate_reader_document=self._get_plate_reader_document(
+                    data, read_type, header, wavelength, plate_well_count
+                ),
+                device_system_document=DeviceSystemDocument(
+                    device_identifier="N/A",
+                    model_number="N/A",
+                    product_manufacturer="BMG LABTECH",
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=filename,
+                    UNC_path=header.path,
+                    software_name="BMG MARS",
+                    ASM_converter_name=self.get_asm_converter_name(),
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+            ),
+            field_asm_manifest="http://purl.allotrope.org/manifests/plate-reader/BENCHLING/2023/09/plate-reader.manifest",
+        )
+
+    def _get_read_type(self, lines: list[str]) -> ReadType:
+        read_type = assert_not_none(
+            re.search(RE_READ_TYPE, "\n".join(lines), flags=re.MULTILINE),
+            msg="Read type not found.",
+        ).group(0)
+        patterns = {
+            "Absorbance": ReadType.ABSORBANCE,
+            "Fluorescence": ReadType.FLUORESCENCE,
+        }
+
+        for key in patterns:
+            if key in read_type:
+                read_type_key = patterns[key]
+        return read_type_key
+
+    def _get_device_control_aggregate_document(
+        self,
+        wavelength: Wavelength,
+        read_type: ReadType,
+    ) -> DeviceControlAggregateDocument:
+
+        if read_type == ReadType.ABSORBANCE:
+            return UltravioletAbsorbancePointDetectionDeviceControlAggregateDocument(
+                device_control_document=[
+                    UltravioletAbsorbancePointDetectionDeviceControlDocumentItem(
+                        device_type="absorbance detector",
+                        detection_type="absorbance",
+                        detector_wavelength_setting=safe_value(
+                            TQuantityValueNanometer,
+                            wavelength.wavelength if wavelength.wavelength else None,
+                        ),
+                    )
+                ]
+            )
+
+        elif read_type == ReadType.FLUORESCENCE:
+            return FluorescencePointDetectionDeviceControlAggregateDocument(
+                device_control_document=[
+                    FluorescencePointDetectionDeviceControlDocumentItem(
+                        device_type="fluorescence detector",
+                        detection_type="fluorescence",
+                        detector_wavelength_setting=safe_value(
+                            TQuantityValueNanometer,
+                            wavelength.wavelength if wavelength.wavelength else None,
+                        ),
+                        excitation_wavelength_setting=safe_value(
```

Comment:
The `safe_value` check makes sense here, but feels overcomplicated since we have the if/else both here and in the function, why not just `TQuantityValueNanometer(wavelength.ex_wavelength) if wavelength.ex_wavelength else None`

Reply:
Simplified detector_wavelength_setting for UV and Fluorescence